src/main/java/qna/domain/Answer.java

```diff
@@ -82,47 +95,54 @@ public Question question() {
 
 	public void question(Question question) {
 		this.question = question;
-	}
-
-	public String contents() {
-		return contents;
+		updatedAtNow();
 	}
 
 	public boolean isOwner(User writer) {
 		return this.writer.equals(writer);
 	}
 
+	public void validateIsOwner(User loginUser) throws CannotDeleteException {
+		if (!isOwner(loginUser)) {
+			throw new CannotDeleteException("사용자와 답변의 작성자가 일치하지 않습니다.");
+		}
+	}
+
 	public boolean isDeleted() {
-		return deleted;
+		return deleted.value();
 	}
 
 	public void delete() {
```

Comment:
```suggestion
	public DeleteHistory delete(User user) {
```

삭제한 이력을 리턴하도록 만들면 어떨까요

Reply:
> 삭제 히스토리를 생성하는 메서드를 따로 두지 말고 delete 결과로 이력을 반환하도록 만들면 어떨까요
> 지금 구조는 클라이언트에서 히스토리 생성까지 챙겨야 합니다.
> (아래 남겨 주신 코멘트를 가져왔습니다.)

이 내용이 참 생각을 많이 하게 했습니다. 🤔 
도메인을 작성하는데 저도 모르게 이런 생각을 가진 것 같습니다. 😒 
1. 도메인은 각자의 고유한 기능을 정의하도록 하며 
2. 도메인 간 서로 의존하지 않는 형태로 작성했습니다.

그러나 이 생각이 잘못되었던 것이,(객체 지향적으로 생각하면)
상속(extends)받고 인터페이스를 구현(implements)하기 때문에 **"고유"** 할 수 없습니다.
오히려 도메인과의 **"관계🎀 를 잘 정의"** 하는 것이 중요한 포인트인 것 같습니다. 

좋은 코멘트 감사합니다!!! 👍 🙇 🥇